PukingPredator/Assets/Scripts/SceneTransition.cs

```diff
@@ -0,0 +1,75 @@
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+using System.Collections;
+
+public class SceneTransition : SingletonMonobehaviour<SceneTransition>
+{
+    [SerializeField]
+    private Image fadePrefab;
+
+    private Image fadeImage;
+
+    private string nextSceneName;
+
+    public void FadeToScene(string sceneName)
+    {
+        Canvas canvas = FindObjectOfType<Canvas>();
+        if (canvas)
+        {
+            fadeImage = Instantiate(fadePrefab, canvas.transform);
+
+            nextSceneName = sceneName;
+            StartCoroutine(FadeIn());
+        } else
+        {
+            SceneManager.LoadScene(sceneName);
+        }
+    }
+
+    private IEnumerator FadeOut()
```

Comment:
The names are currently the opposite of what i generally see (you fade out of the current scene and then fade into the target scene, not fade into the transition and then out of the transition). It also should have documentation regardless so you dont need to read the code to figure out which it is doing